webapp/components/survey/SurveyForm/nodeDefs/components/types/nodeDefEntityFormGrid.js

```diff
@@ -32,6 +33,16 @@ const NodeDefEntityFormGrid = (props) => {
     }, 200)
   }, [])
 
+  let childDefNamesOrdered = null
+  if (ProcessUtils.isEnvDevelopment) {
+    // ordered child def names used in tests
+    const layoutChildrenItemIdsOrdered = NodeDefLayout.getLayoutChildrenItemIdsOrdered(surveyCycleKey)(nodeDef)
+    childDefNamesOrdered = layoutChildrenItemIdsOrdered.map((childDefUuid) => {
+      const childDef = childDefs.find((def) => NodeDef.getUuid(def) === childDefUuid)
```

Comment:
can avoid the iteration by getting the nodeDef with Survey.getNodeDefByUuid ?
but, can't you set the uuids directly as data-child-defs ?

Reply:
also, call the method NodeDefLayout.getLayoutChildrenUuids. add a jsdoc comment that they are ordered

Reply:
ok